src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -573,19 +573,9 @@ public enum CompareExchangeTombstonesCleanupState
 
         private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEtagToDelete(TransactionOperationContext context, string databaseName, DatabaseObservationState state, out long maxEtag)
         {
-            List<long> periodicBackupTaskIds;
-            maxEtag = long.MaxValue;
-
-            if (state?.RawDatabase != null)
-            {
-                periodicBackupTaskIds = state.RawDatabase.PeriodicBackupsTaskIds;
-            }
-            else
-            {
-                using (var rawRecord = _server.Cluster.ReadRawDatabaseRecord(context, databaseName))
-                    periodicBackupTaskIds = rawRecord.PeriodicBackupsTaskIds;
-            }
+            maxEtag = -1;
```

Comment:
I changed the default maxEtag to be on the safe side since deleting tombstones prematurely can cause later requests to hang.

Additionally, while before the PR there could be a situation where no one subscribed to the compare exchange cleaner (no backup and no indexes, which is rare but theoretically possible), now there must be at least the cluster-wide transaction, and it should be updated at least at a defined interval.

There is also an issue with clusters that have one node with a version from before the PR. This node can trigger tombstone deletion while the cluster-wide transaction hasn't handled the tombstone yet (since this node is not familiar with the new logic). This can cause all requests to hang. However, this is a very edge case and requires multiple conditions:

The last compare exchange modification is a deletion.
The leader is the old version node.
The tombstone cleaner cleans the tombstone before the cluster-wide transaction handles it.
No more compare exchange modifications or cluster-wide transactions occur.
I spoke with @karmeli87, and we decided that since this is a very edge case, almost hypothetical, we can accept it.

Also, if the leader is a new version node, it will always get 0 from the old version node and so will never delete the tombstone. I can fix that easily, but I prefer not to delete on missing information rather than implement logic to ignore that information. Additionally, the mixed cluster should be a temporary state.